refs/heads/master: internal/metamorphic/crossversion/crossversion_test: TestMetaCrossVersion failed

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/21252097?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/21252097?buildTab=artifacts#/) on refs/heads/master @ [1454d2bc0f37](https://github.com/cockroachdb/pebble/commits/1454d2bc0f37):


```
                --- PASS: TestMeta/execution/random-007 (1.88s)
                --- PASS: TestMeta/execution/standard-022 (0.84s)
                --- PASS: TestMeta/execution/standard-024 (0.85s)
                --- PASS: TestMeta/execution/standard-023 (0.80s)
                --- PASS: TestMeta/execution/random-016 (1.39s)
                --- PASS: TestMeta/execution/random-012 (1.61s)
                --- PASS: TestMeta/execution/random-005 (2.26s)
                --- PASS: TestMeta/execution/standard-009 (0.94s)
                --- PASS: TestMeta/execution/standard-016 (1.14s)
                --- PASS: TestMeta/execution/random-024 (2.12s)
                --- PASS: TestMeta/execution/random-026 (1.83s)
                --- PASS: TestMeta/execution/random-028 (2.25s)
                --- FAIL: TestMeta/execution/random-025 (2.33s)
                --- PASS: TestMeta/execution/standard-013 (0.78s)
                --- PASS: TestMeta/execution/random-027 (2.78s)
                --- PASS: TestMeta/execution/standard-014 (0.94s)
                --- PASS: TestMeta/execution/standard-011 (0.89s)
                --- PASS: TestMeta/execution/standard-012 (1.09s)
                --- PASS: TestMeta/execution/random-029 (3.21s)
                --- PASS: TestMeta/execution/standard-010 (0.81s)
                --- PASS: TestMeta/execution/standard-020 (0.84s)
                --- PASS: TestMeta/execution/random-022 (0.95s)
                --- PASS: TestMeta/execution/random-023 (1.18s)
                --- PASS: TestMeta/execution/standard-019 (0.86s)
                --- PASS: TestMeta/execution/random-021 (1.24s)
                --- PASS: TestMeta/execution/standard-018 (0.84s)
                --- PASS: TestMeta/execution/random-015 (1.77s)
                --- PASS: TestMeta/execution/standard-021 (1.80s)
                --- PASS: TestMeta/execution/standard-005 (0.89s)
                --- PASS: TestMeta/execution/standard-007 (0.80s)
                --- PASS: TestMeta/execution/standard-008 (1.16s)
                --- PASS: TestMeta/execution/random-000 (1.87s)
                --- PASS: TestMeta/execution/standard-028 (0.68s)
                --- PASS: TestMeta/execution/standard-006 (0.94s)
                --- PASS: TestMeta/execution/random-001 (2.21s)
                --- PASS: TestMeta/execution/random-002 (2.25s)
                --- PASS: TestMeta/execution/standard-029 (0.86s)
                --- PASS: TestMeta/execution/standard-003 (0.96s)
                --- PASS: TestMeta/execution/random-003 (3.13s)
                --- PASS: TestMeta/execution/standard-002 (0.93s)
                --- PASS: TestMeta/execution/standard-004 (0.91s)
                --- PASS: TestMeta/execution/standard-001 (0.66s)
                --- PASS: TestMeta/execution/random-020 (1.86s)
                --- PASS: TestMeta/execution/random-019 (2.09s)
                --- PASS: TestMeta/execution/standard-027 (0.64s)
                --- PASS: TestMeta/execution/random-018 (1.69s)
                --- PASS: TestMeta/execution/standard-015 (10.23s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-master/sha=584cd30f-seed=17689493677796644572-opts=random-013(sha=3dc86d8b-seed=11580307797067454614-opts=standard-019(sha=16881147-seed=17808775994342277341-opts=random-004(sha=8b6bc41b-seed=13812030803756442294-opts=standard-006(<empty>)))) (80.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: PEBBLE-1368

--- a/table_stats.go
+++ b/table_stats.go
@@ -1061,6 +1061,7 @@ func newCombinedDeletionKeyspanIter(
 	mIter.Init(comparer, transform, new(keyspanimpl.MergingBuffers))
 	iter, err := r.NewRawRangeDelIter(ctx, m.FragmentIterTransforms(), env)
 	if err != nil {
+		mIter.Close()
 		return nil, err
 	}
 	if iter != nil {
@@ -1112,6 +1113,7 @@ func newCombinedDeletionKeyspanIter(
 
 	iter, err = r.NewRawRangeKeyIter(ctx, m.FragmentIterTransforms(), env)
 	if err != nil {
+		mIter.Close()
 		return nil, err
 	}
 	if iter != nil {

